mindsdb/api/http/namespaces/agents.py

```diff
@@ -239,6 +241,136 @@ def delete(self, project_name, agent_name):
+    db.session.commit()
+    db.session.flush()
+
+    completion_stream = session.agents_controller.get_completion_stream(
+        existing_agent,
+        messages,
+        trace_id=trace_id,
+        observation_id=observation_id,
+        project_name=project_name,
+        tools=[],
+    )
+    last_output = None
+    for chunk in completion_stream:
+        chunk_obj = {}
+        if 'output' in chunk:
+            # Langchain final output.
+            chunk_obj['output'] = chunk['output']
+        if 'messages' in chunk:
+            # Langchain messages in output/actions.
+            chunk_obj['messages'] = [{'content': m.content} for m in chunk['messages']]
+        if 'actions' in chunk:
+            # Langchain actions.
+            chunk_obj['actions'] = [{
+                'tool': a.tool,
+                'tool_input': a.tool_input,
+                'log': a.log
+            } for a in chunk['actions']]
+        if 'steps' in chunk:
+            # Langchain steps (similar to actions).
+            chunk_obj['steps'] = [{'observation': s.observation} for s in chunk['steps']]
+        chunk_str = json.dumps(chunk_obj)
+        # Stream parsed & formatted Langchain streaming chunk.
+        yield 'data: {}\n\n'.format(chunk_str)
+        if 'output' in chunk:
+            last_output = chunk_obj
+    if run_completion_span is not None and api_trace is not None:
+        run_completion_span.end(output=last_output)
+        api_trace.update(output=last_output)
+
+
+@ns_conf.route('/<project_name>/agents/<agent_name>/completions/stream')
+@ns_conf.param('project_name', 'Name of the project')
+@ns_conf.param('agent_name', 'Name of the agent')
+class AgentCompletionsStream(Resource):
+    @ns_conf.doc('agent_completions_stream')
+    @api_endpoint_metrics('POST', '/agents/agent/completions/stream')
+    def post(self, project_name, agent_name):
+        # Check for required parameters.
+        if 'messages' not in request.json:
+            return http_error(
+                HTTPStatus.BAD_REQUEST,
+                'Missing parameter',
+                'Must provide "messages" parameter in POST body'
+            )
+        session = SessionController()
+        try:
+            existing_agent = session.agents_controller.get_agent(agent_name, project_name=project_name)
+            if existing_agent is None:
+                return http_error(
+                    HTTPStatus.NOT_FOUND,
+                    'Agent not found',
+                    f'Agent with name {agent_name} does not exist'
+                )
+        except ValueError:
+            # Project needs to exist.
+            return http_error(
+                HTTPStatus.NOT_FOUND,
+                'Project not found',
+                f'Project with name {project_name} does not exist'
+            )
+
+        # Model needs to exist.
+        model_name_no_version, version = db.Predictor.get_name_and_version(existing_agent.model_name)
```

Comment:
To be able to use agent without model in restapi: I removed this block today in https://github.com/mindsdb/mindsdb/pull/9481